compiler-plugin/src/test/kotlin/arrow/meta/plugins/typeclasses/TypeClassesTest.kt

```diff
@@ -0,0 +1,83 @@
+package arrow.meta.plugins.typeclasses
+
+import arrow.meta.plugin.testing.CompilationTest
+import org.junit.Test
+
+class TypeClassesTest : CompilationTest {
+
+  @Test
+  fun `simple_case`() {
+    """
+    | import arrow.Kind
+    | import arrow.given
+    | import arrow.core.Some
+    | import arrow.core.Option
+    | import arrow.extension
+    | import arrow.core.ForOption
+    | import arrow.core.fix
+    | import arrow.core.None
+    |
+    | //metadebug
+    |
+    | @extension
+    | object OptionMappable : Mappable<ForOption> {
+    |   override fun <A, B> Kind<ForOption, A>.map(f: (A) -> B): Kind<ForOption, B> =
+    |     when (val o: Option<A> = this.fix()) {
+    |       is Some -> Some(f(o.t))
+    |       None -> None
+    |     }
+    | } 
+    | 
+    | interface Mappable<F> {
+    |   fun <A, B> Kind<F, A>.map(f: (A) -> B): Kind<F, B>
+    | }
+    |
+    | object Test {
```

Comment:
Do we need to use a wrapping `object` class for testing reasons?

Reply:
No, that `Test` is part of the code snippet under test. Maybe the use of `Test` is confusing if it's inside the code snippet... Good point, thanks! :+1: